app/src/main/java/com/example/chatbotandroid/utils/Constants.kt

```diff
@@ -0,0 +1,8 @@
+package com.example.chatbotandroid.utils
+
+object Constants {
+    val SEND_TYPE = "SENDER"
+    val RECEIVE_TYPE = "RECEIVER"
+    //Must connect to 10.0.2.2 to connect to local network from emulator
+    val WEB_SOCKET_URL = "ws://10.0.2.2:8080/chat"
```

Comment:
one thing worth thinking about is: if we wanted to write some automated tests for this app, how would we make that work? e.g. what if we wanted to write an automated test where a message is typed out and the 'send' button is clicked, and we asserted that a message showed up -- but without actually being connected to a web socket. it's not an easy thing to do, but the answer lies in a combination of dependency injection and depending on abstractions. 

one first step would be making MainActivity totally unaware of anything like what a web socket is, or how to make a connection on one. if MainActivity just had a handle on some opaque object called like "ChatServer" and could call a function on it that returns a stream of Messages (and other relevant functions), then that'd be a nice abstraction to hide an underlying web socket implementation. in a test environment, that "ChatServer" use could have a fake, in-memory implementation that gets swapped in.

Reply:
Agreed! Something I've done in the past is follow what I believe is called Factory architecture, then we can create MockClients that could fake data send and we could write automated tests. Really nice comment! :)